### PR TITLE
refactor(circuits): refactor big comparison component

### PR DESCRIPTION
**Pull request by vimwitch**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/circuits/circuits/bigComparators.circom [hunk 1]
```diff
@@ -3,51 +3,47 @@ include "./circomlib/circuits/bitify.circom";
...
+    low_lt.in[0] <== low[0].out;
+    low_lt.in[1] <== low[1].out;
 
     component mux = Mux1();
-    mux.s <== quotient_lt.out;
+    mux.s <== high_lt.out;
 
-    mux.c[0] <== remainder_lt.out;
+    mux.c[0] <== low_lt.out;
     mux.c[1] <== 1;
```
Comment (vivianjeng) [inline packages/circuits/circuits/bigComparators.circom:44]: This should be wrong.
If `high[0] >= high[1]`, then `high_lt.out = 0`
then we should check if `high[0] == high[1]` and then compare `low[0]` and `low[1]`
or it should return `0` rather than `low_lt.out`

**Review by vivianjeng — CHANGES_REQUESTED** [review]
we should add some tests for this to compare many big numbers

File: packages/circuits/circuits/bigComparators.circom (hunk 1 above)
Comment (vimwitch) [reply packages/circuits/circuits/bigComparators.circom:44]: Ah you're right, lower order bits should only be checked if `high[0] == high[1]`, otherwise it returns 0. I agree, this needs to be tested around the edge boundaries.

File: packages/circuits/package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
     ],
     "scripts": {
         "pre": "node scripts/prepare",
+        "buildTest": "TEST=1 yarn build",
```
Comment (vivianjeng) [inline packages/circuits/package.json:22]: I think it can be executed before `test`?
So the test script become `yarn buildTest && mocha -r ts-node/register test/*.test.ts --exit`

**Review by vivianjeng — COMMENTED** [review]
good tests 👍🏻

File: packages/circuits/package.json (hunk 2 above)
Comment (vimwitch) [reply packages/circuits/package.json:22]: Ah, i prefer not. I often test using `yarn test --grep 'test name'`, so i might not want to build all circuits.

Comment (vivianjeng) [reply packages/circuits/package.json:22]: but it would fail if developers clone can run `yarn build` and `yarn test`
we should either skip these tests locally without building circuits or running these tests and building circuits.

Comment (vimwitch) [reply packages/circuits/package.json:22]: Sure, maybe build all circuits by default and remove the test only circuits as a part of the `pre` script

Comment (vimwitch) [reply packages/circuits/package.json]: [c7bf309](https://github.com/Unirep/Unirep/pull/313/commits/c7bf3099bb1bd7229622f7353c1223bfd61d0f54)

File: packages/circuits/scripts/prepare.js [hunk 3]
```diff
@@ -31,6 +31,12 @@ try {
     )
 } catch (_) {}
 
+// remove test keys
+const zkPath = path.join(__dirname, '../dist/zksnarkBuild')
+fs.readdirSync(zkPath)
+    .filter((filename) => /^bigComparators/.test(filename))
```
Comment (vivianjeng) [inline packages/circuits/scripts/prepare.js:37]: it would be better if we set the test circuits in a list if we have many test circuits in the future.
but it would be fine to be in this PR.

**Review by vivianjeng — APPROVED** [review]
LGTM
